src/test/java/io/pinecone/clients/ConnectionsMapTest.java

```diff
@@ -0,0 +1,134 @@
+package io.pinecone.clients;
+
+import io.pinecone.configs.PineconeConfig;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+import org.openapitools.client.ApiException;
+import org.openapitools.client.api.ManageIndexesApi;
+import org.openapitools.client.model.IndexModel;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.Mockito.*;
+import static org.junit.jupiter.api.Assertions.*;
+
+public class ConnectionsMapTest {
+
+    @Test
+    public void testMultiplePineconeInstancesForIndexes() {
+        String host = "some-host";
+        ManageIndexesApi manageIndexesApi = Mockito.mock(ManageIndexesApi.class);
+        IndexModel indexModel = new IndexModel();
+        indexModel.setHost(host);
+
+        PineconeConfig config = new PineconeConfig("testApiKey");
+        config.setHost(host);
+        Pinecone pinecone1 = new Pinecone(config, manageIndexesApi);
+        Pinecone pinecone2 = new Pinecone(config, manageIndexesApi);
+
+        when(pinecone1.describeIndex("index1")).thenReturn(indexModel);
+        when(pinecone1.describeIndex("index2")).thenReturn(indexModel);
+        when(pinecone2.describeIndex("index1")).thenReturn(indexModel);
+        when(pinecone2.describeIndex("index2")).thenReturn(indexModel);
+
+        Index index1Pinecone1 = pinecone1.getIndexConnection("index1");
+        Index index1Pinecone2 = pinecone1.getIndexConnection("index2");
+        Index index2Pinecone1 = pinecone2.getIndexConnection("index1");
+        Index index2Pinecone2 = pinecone2.getIndexConnection("index2");
+        assertEquals(pinecone1.getConnectionsMap().size(), 2);
+        assertEquals(pinecone2.getConnectionsMap().size(), 2);
+
+        index1Pinecone1.close();
```

Comment:
Should you just put this all in a try with resources loop?

Reply:
Yes that's another option, it'll close all of the connections